codegen/src/main/java/software/amazon/awssdk/codegen/poet/builder/AsyncClientBuilderClass.java

```diff
@@ -121,16 +121,16 @@ private MethodSpec endpointProviderMethod() {
     private MethodSpec buildClientMethod() {
         return MethodSpec.methodBuilder("buildClient")
                          .addAnnotation(Override.class)
-                         .addModifiers(Modifier.PROTECTED)
+                         .addModifiers(Modifier.PROTECTED, Modifier.FINAL)
                          .returns(clientInterfaceName)
                          .addStatement("$T clientConfiguration = super.asyncClientConfiguration()", SdkClientConfiguration.class)
                          .addStatement("this.validateClientOptions(clientConfiguration)")
                          .addStatement("$T endpointOverride = null", URI.class)
                          .addCode("if (clientConfiguration.option($T.ENDPOINT_OVERRIDDEN) != null"
-                                  + "&& clientConfiguration.option($T.ENDPOINT_OVERRIDDEN)) {"
+                                  + "&& clientConfiguration.option($T.ENDPOINT_OVERRIDDEN) == $T.TRUE) {"
```

Comment:
Sorry, I updated my example. `Boolean.TRUE.equals(clientConfiguration.option($T.ENDPOINT_OVERRIDDEN))` is safer/more correct. Can we use that instead?

Reply:
+1